### ISSUE TITLE
messages with `msg.reset` is allowed through the delay node 

### ISSUE THREAD

**Issue by marcus-j-davies** [node-red/node-red#4337]
### Current Behavior

When the delay node is configured as.

 - Rate Limit
 
 And
 
  - No messages are currently held back.
  

Messages containing `msg.reset` are being passed through 

### Expected Behavior

It appears this is new behaviour, and previously, `msg.reset` has never been passed through, as it should be a control message.

I'm unsure if this is now expected behaviour, or something has gone pop! 

### Steps To Reproduce

 - Set Delay node to `Rate Limit`
 - send it a message with `msg.reset`
 
 See that it passes the messages through.

### Example flow

```
[{"id":"847f3a065f5f8cdd","type":"tab","label":"Flow 1","disabled":false,"info":"","env":[]},{"id":"b0f341d4815fa7ea","type":"debug","z":"847f3a065f5f8cdd","name":"debug 4","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":1115,"y":100,"wires":[]},{"id":"a4a9b730b905294c","type":"delay","z":"847f3a065f5f8cdd","name":"","pauseType":"rate","timeout":"5","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"hour","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":true,"allowrate":false,"outputs":1,"x":910,"y":100,"wires":[["b0f341d4815fa7ea"]]},{"id":"351578f6dd252a3f","type":"inject","z":"847f3a065f5f8cdd","name":"reset","props":[{"p":"reset","v":"true","vt":"bool"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":680,"y":220,"wires":[["a4a9b730b905294c"]]},{"id":"c944b4bae18a02ea","type":"inject","z":"847f3a065f5f8cdd","name":"","props":[{"p":"payload"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":680,"y":100,"wires":[["a4a9b730b905294c"]]}]
```

### Environment

- Node-RED version: 3.1.0 (but I cant help thinking I noticed this change a little before)

Not at my workstation currently, but this is discussed below also:

https://discourse.nodered.org/t/has-something-changed-with-the-delay-node/81221

**Comment by jberstler** [comment]
I'm seeing this as well on 3.1.0, and can't help but wonder if it is related to the change introduced in #4203 

In my case, this behavior change is especially disastrous as I have flows that loop back to a delay (just passing along whatever `msg` happens to be, including the `reset`) which results in an infinite loop now that those messages are getting passed through the delay node instead of getting stopped as before.

```
[{"id":"fabade9e801b8108","type":"tab","label":"Flow 1","disabled":false,"info":"","env":[]},{"id":"53b79d00324a59f0","type":"delay","z":"fabade9e801b8108","name":"","pauseType":"rate","timeout":"10","timeoutUnits":"minutes","rate":"1","nbRateUnits":"10","rateUnits":"minute","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":true,"allowrate":false,"outputs":1,"x":660,"y":320,"wires":[["550278e216ba0ac6"]]},{"id":"aa21cf6786e86722","type":"inject","z":"fabade9e801b8108","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":260,"y":320,"wires":[["88d837e038616a61"]]},{"id":"88d837e038616a61","type":"change","z":"fabade9e801b8108","name":"","rules":[{"t":"set","p":"reset","pt":"msg","to":"true","tot":"bool"}],"action":"","property":"","from":"","to":"","reg":false,"x":440,"y":320,"wires":[["53b79d00324a59f0"]]},{"id":"550278e216ba0ac6","type":"function","z":"fabade9e801b8108","name":"function 1","func":"// dummy node that just passes along msg\nreturn msg;","outputs":1,"timeout":0,"noerr":0,"initialize":"","finalize":"","libs":[],"x":860,"y":320,"wires":[["dd8dc66cbf79f62e","073f54426ebbb8a8"]]},{"id":"073f54426ebbb8a8","type":"debug","z":"fabade9e801b8108","name":"debug 9","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":1060,"y":200,"wires":[]},{"id":"dd8dc66cbf79f62e","type":"change","z":"fabade9e801b8108","name":"","rules":[{"t":"delete","p":"reset","pt":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":1090,"y":320,"wires":[["53b79d00324a59f0"]]},{"id":"40de19d90e4ef76e","type":"comment","z":"fabade9e801b8108","name":"Gets called twice - shouldn't get called at all","info":"","x":1170,"y":160,"wires":[]},{"id":"47d1bf458d1a3549","type":"comment","z":"fabade9e801b8108","name":"Be careful! Without deleting msg.reset, this is an infinite loop.","info":"","x":1220,"y":280,"wires":[]}]
```

**Comment by marcus-j-davies** [comment]
Nice example flow @jberstler to demo the effect.
I'm going to ping @dceejay to get his input.

**Comment by knolleary** [comment]
Fixed since 3.1.1
